File: chapter_2/exercise_1/hello/lib/hello.ex
defmodule Hello do
  @moduledoc """
  Documentation for Hello.
  """

  def hello do
    get_name()
    |> build_output
    |> IO.puts
  end

  def get_name do
    IO.gets("What is your name? ")
    |> String.trim
  end

  def build_output(name) do
    "Hello, #{name}, nice to meet you!"
  end
end

File: chapter_2/exercise_1/hello.c
#include <stdio.h>
#include <string.h>

int main()
{
  int max_name = 64;
  char name[max_name];

  printf("What is your name? ");
  fgets(name, max_name - 1, stdin);

  name[strcspn(name, "\r\n")] = 0; // alter the string to terminate at the newline

  printf("Hello, %s, nice to meet you!\n", name);
}
